src/forms/FormFieldAdvancedSelection/AdvancedSelectionDrawer.tsx

```diff
@@ -43,12 +43,38 @@ const AdvancedSelectionDrawer = (props: AdvanceSelectionDrawerPropTypes): ReactE
 	const [canLoadMore, setCanLoadMore] = useState<boolean>(true);
 	const [filteredOptions, setFilteredOptions] = useState<MosaicLabelValue[]>([]);
 	const [checkboxListDisabled, setCheckboxListDisabled] = useState(fieldDef.disabled);
+	const [topComponentDrawerHeight, setTopComponentDrawerHeight] = useState("0px");
+	const [formLayoutPadding, setFormLayoutPadding] = useState("0px");
 
 	const chipListRef:{ current?: HTMLDivElement } = useRef();
+	const loadMoreButtonRef:{ current?: HTMLDivElement } = useRef();
+	const searchInputRef:{ current?: HTMLDivElement } = useRef();
 
 	const { state, dispatch } = useForm();
 
-	const chipListHeight: number = chipListRef?.current?.offsetHeight ? chipListRef?.current?.offsetHeight + 30 : 0;
+	const chipListHeight: string = chipListRef?.current?.offsetHeight

```

Comment:
This is a good approach, I just have some observations to optimize memory and prevent re-renders:
1. The value could be stored using a useMemo with the ref as dependency, that way it won't re-calculate with every other change to the component.
2. Line 56 doesn't need all the question marks because you're already making sure the object exists one line above, so you could just have it like this: ${chipListRef.current.offsetHeight}

These comments go to all other variables you defined under this.